src/ILLink.Shared/TrimAnalysis/NullableRuntimeTypeHandleWithGenericParameterValue.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using ILLink.Shared.DataFlow;
+using ILLink.Shared.TypeSystemProxy;
+
+namespace ILLink.Shared.TrimAnalysis
+{
+	/// <summary>
+	/// This represents a type handle of a Nullable<T> where T is an unknown value with DynamicallyAccessedMembers annotations. 
+	/// It is necessary to track the underlying type to ensure DynamicallyAccessedMembers annotations on the underlying type match the target parameters where the Nullable is used.
+	/// </summary>
+	sealed record NullableRuntimeTypeWithDamHandleValue : SingleValue
+	{
+		public NullableRuntimeTypeWithDamHandleValue (in TypeProxy nullableType, in SingleValue underlyingTypeValue)
```

Comment:
`UnderlyingTypeValue needs to be SingleValue here because in MethodBodyScanner, a RuntimeTypeHandleForGenericParameterValue is passed as the underlying type.